base_layer/core/src/types.rs

```diff
@@ -77,3 +79,12 @@ lazy_static! {
     pub static ref PROVER: RangeProofService =
         RangeProofService::new(MAX_RANGE_PROOF_RANGE, &COMMITMENT_FACTORY).unwrap();
 }
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct RangeProof(pub Vec<u8>);
```

Comment:
Creating a new struct here is fine; but having a public inner type leads to an unsavoury API. Seeing `proof.0` everywhere isn't cool.

I'd suggest making the inner type private and implementing the `ByteArray` trait  so that you expose `as_bytes`.